### PR TITLE
[DI] Add samples and tests

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

**Comment by YalinLi0312** [comment]
Hi @iscai-msft , there are some LROs return empty part from deserialized result so that we have to patch them. Can you help to review if that's expected?

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py:1]: Why did these operations need to be patched?

Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py:1]: Because they are returning wrong section of the deserialized result, the return value in generated operations are `None`.

File: sdk/documentintelligence/azure-ai-documentintelligence/tests/testcase.py
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/tests/testcase.py:136]: This file will also need some clean up since it's referencing variables not used in this new library.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout.py
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_layout.py:13]: I dont recall if the service still returns all of these elements in the layout model anymore. Worth taking a look to make sure later.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py:50]: I suppose we'll work to fix this for the next release? So customers dont have to specify the content type when streaming the file

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_receipts_from_url.py
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_receipts_from_url.py:17]: These links should really be updated to say document intelligence and not form recognizer anymore. Wont block on this now though

**Review by catalinaperalta — COMMENTED** [review]
There is some general clean up to be done in this library, let's make sure before the next release that we arent carrying over code that doesnt belong in this new library.

In this PR we should remove the sample forms that arent actually used. Also fix the naming on that test-resources-post script if you need it.

I think you also need the samples README, but it may not be a blocker for this release, I'm checking on that.

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py
Comment (catalinaperalta) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py:1]: Can you point me to the line you had to modify? I tried to compare with what was generated but I think it would be faster if you just let me know what line was changed

**Comment by catalinaperalta** [comment]
CI says the README is missing several sections also FYI

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py
Comment (catalinaperalta) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py:1]: Ok so I saw that the line that was changed in the patch is this one: https://github.com/Azure/azure-sdk-for-python/blob/9fa59eb33aa9b66532e95d8ca57a44c137b1122a/sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py#L65


I want to make sure we're getting the right model returned for these operations. You need to have at least 1 test for each of your patched operations and make some asserts to ensure that deserialization was correct. We need guarantee that these patched operations are indeed returning the right information for the model we say is going to be returned.

Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py:1]: - You found the right place. 
- yup, I have tests for each. That's how I found the operations not working.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_forms/labeled_tables/dynamic/label_table_dynamic_rows1.pdf
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_forms/labeled_tables/dynamic/label_table_dynamic_rows1.pdf:1]: I didn't check, I copied all data from form-recognizer.

File: sdk/documentintelligence/test-reaources-post.ps1
Comment (YalinLi0312) [reply sdk/documentintelligence/test-reaources-post.ps1:1]: It's from form-recognizer. I don't see it in other package, looks it's not required. Will remove.

File: sdk/documentintelligence/test-resources.json
Comment (YalinLi0312) [reply sdk/documentintelligence/test-resources.json:319]: yup, that's the plan.

File: sdk/documentintelligence/azure-ai-documentintelligence/tests/testcase.py
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/tests/testcase.py:136]: Agree. I think we can leave the clean up (and in other files) later as the releasing is expect to ship today.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/sample_analyze_read.py:50]: Right, it's a bug need to fix in python generator.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/README.md
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/README.md:18]: Let's just simplify this file for today's release and remove this table. Add a link to the samples directory instead. Open an issue for yourself to fix this before the next release.

File: sdk/documentintelligence/azure-ai-documentintelligence/README.md
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/README.md:47]: This comment should be addressed for a future release:

We should verify if these are still the recommended commands for creating a resource.

File: sdk/documentintelligence/azure-ai-documentintelligence/README.md
Comment (catalinaperalta) [inline sdk/documentintelligence/azure-ai-documentintelligence/README.md:285]: Wasnt this model deprecated in the current service API version?

**Review by catalinaperalta — COMMENTED** [review]
Most of the comments that are still unaddressed on this PR can be added in a future release. Let's prioritize the following to get this PR merged in today to unblock your release:
- Skip the 4 failing tests
- Fix ci Analyze errors:
   -  there are mypy and pylint ci errors in the analyze step

File: sdk/documentintelligence/azure-ai-documentintelligence/README.md
Comment (kinelski) [reply sdk/documentintelligence/azure-ai-documentintelligence/README.md:285]: I can confirm it was deprecated. We get a 404 if we try to use it with the new version.

Comment (catalinaperalta) [reply sdk/documentintelligence/azure-ai-documentintelligence/README.md:285]: @YalinLi0312 we shouldnt be recommending a deprecated model to users in the README

**Comment by YalinLi0312** [comment]
> Most of the comments that are still unaddressed on this PR can be added in a future release. Let's prioritize the following to get this PR merged in today to unblock your release:
> 
> * Skip the 4 failing tests
> * Fix ci Analyze errors:
>   
>   * there are mypy and pylint ci errors in the analyze step

- The 4 failing tests are easy to fix, already updated.
- For mypy, as there are 2 errors in generated code, I would skip the check for now. Pylint errors are fixed in this PR 
- I leave cleanup stuff to next release.

**Review by catalinaperalta — APPROVED** [review]
Please remove the prebuilt-document example from the readme.

File: sdk/documentintelligence/azure-ai-documentintelligence/README.md
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/README.md]: Removed. `prebuilt-document` and `prebuilt-businessCard` are deprecated.

**Pull request merged**
